parts/k8s/kubernetesmastercustomdata.yml

```diff
@@ -113,13 +117,15 @@ write_files:
     RemainAfterExit=yes
     ExecStart=/usr/local/bin/health-monitor.sh container-runtime
 
+{{if IsUbuntu1604 .MasterProfile.Distro}}
```

Comment:
@jackfrancis does skipping this for 18.04 make sense?

Reply:
yes

Not to up-scope your work, but a more sustainable evolution of this would be add a "needsProposedPackages" func that we can maintain over time. The first implementation of that would be what you have here: basically "am I 1604? then I need to install proposed packages."

Reply:
The reason I decided to go with `IsUbuntu1604` is because I am going to need that function for other non-proposed-packages things that are specific to ubuntu1804 (such as the new kubelet config flag) so `IsUbuntu1604` seemed more general, reusable than `needsProposedPackages`, does that make sense?

Reply:
Oh, definitely don't delete the "IsUbuntu1604" func, just have the "needsProposedPackages" invoke it.

Actually, now that I think of it, we should probably just *always* deliver the proposed repo reference for folks who want to explicitly get proposed/<package>. If this is implemented correctly, it has no effect upon any "normal" apt operations. It just wires the OS up so it knows where to get pre-production packages, if so desired (i.e., packages prefixed with "proposed/").

Reply:
Agreed but out of scope of this PR, let's make implementing a configurable "proposedPackages" apimodel config its own task.

Reply:
BTW thinking about this more, making this a configurable option will make it tricky to pre-install in VHD so definitely worth thinking about it as a separate task.

Reply:
I'd just keep it as-is. As mentioned, the purpose of this is to ensure that -proposed isn't used as an option for component upgrade. See here:

https://wiki.ubuntu.com/Testing/EnableProposed

What *does* need to be done is for all "xenial-proposed" references in cloud-init and CSE to be changed to "bionic-proposed".